lxc file mount panic

Both running LXD 5.4.

Remote server is running stock ubuntu 18.04.2 LTS and client is stock Ubuntu 20.04.3 LTS.

```
$ lxc file mount work-stage-machine:fs-design-dev/ tmp-remote/
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x95a130]

goroutine 1 [running]:
github.com/lxc/lxd/client.(*ProtocolLXD).rawSFTPConn(0xc00018a480, 0xc000272510)
	/build/lxd/parts/lxd/src/client/lxd_instances.go:1381 +0x250
github.com/lxc/lxd/client.(*ProtocolLXD).GetInstanceFileSFTPConn(0xc00018a480, {0x7fff866abff3, 0xd})
	/build/lxd/parts/lxd/src/client/lxd_instances.go:1429 +0x12d
main.(*cmdFileMount).sshfsMount(0x0?, {0xd44078, 0xc000028130}, {{0x7fff866abfe0, 0x12}, {0xd4e628, 0xc00018a480}, {0x7fff866abff3, 0xe}}, {0x7fff866abff3, ...}, ...)
	/build/lxd/parts/lxd/src/lxc/file.go:1041 +0xaf
main.(*cmdFileMount).Run(0xc0001f8a80, 0xc00027a500, {0xc000253060?, 0x2, 0x0?})
	/build/lxd/parts/lxd/src/lxc/file.go:1032 +0x492
github.com/spf13/cobra.(*Command).execute(0xc00027a500, {0xc000253000, 0x2, 0x2})
	/build/lxd/parts/lxd/src/.go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:872 +0x694
github.com/spf13/cobra.(*Command).ExecuteC(0xc000248000)
	/build/lxd/parts/lxd/src/.go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:990 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
	/build/lxd/parts/lxd/src/.go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:918
main.main()
	/build/lxd/parts/lxd/src/lxc/main.go:276 +0x1a33
```

--- a/client/util.go
+++ b/client/util.go
@@ -23,7 +23,6 @@ func tlsHTTPClient(client *http.Client, tlsClientCert string, tlsClientKey strin
 	// Define the http transport
 	transport := &http.Transport{
 		TLSClientConfig:       tlsConfig,
-		DialContext:           shared.RFC3493Dialer,
 		Proxy:                 shared.ProxyFromEnvironment,
 		DisableKeepAlives:     true,
 		ExpectContinueTimeout: time.Second * 30,
@@ -37,12 +36,9 @@ func tlsHTTPClient(client *http.Client, tlsClientCert string, tlsClientKey strin
 	}
 
 	// Special TLS handling
-	//lint:ignore SA1019 DialContext doesn't exist in Go 1.13
-	transport.DialTLS = func(network string, addr string) (net.Conn, error) {
+	transport.DialTLSContext = func(ctx context.Context, network string, addr string) (net.Conn, error) {
 		tlsDial := func(network string, addr string, config *tls.Config, resetName bool) (net.Conn, error) {
-			// TCP connection
-			//lint:ignore SA1019 DialContext doesn't exist in Go 1.13
-			conn, err := transport.DialContext(context.Background(), network, addr)
+			conn, err := shared.RFC3493Dialer(ctx, network, addr)
 			if err != nil {
 				return nil, err
 			}

